### PR TITLE
Separate spark module

### PR DESCRIPTION
**Pull request by DarthMax**
Move spark related code into separate spark module

### CODE REVIEW

File: caps-core/pom.xml [hunk 1]
```diff
@@ -11,12 +11,88 @@
   </parent>
 
   <artifactId>caps-core</artifactId>
-  <version>1.0-SNAPSHOT</version>
```
Comment (Mats-SX) [inline caps-core/pom.xml]: Doesn't each artifact need to specify a version?

File: caps-core/pom.xml [hunk 2]
```diff
@@ -11,12 +11,88 @@
...
+      <!--<groupId>org.apache.spark</groupId>-->
+      <!--<artifactId>spark-sql_${project.scala.binary.version}</artifactId>-->
+      <!--<version>${dep.spark.version}</version>-->
+    <!--</dependency>-->
+
+    <!--<dependency>-->
+      <!--<groupId>org.apache.spark</groupId>-->
+      <!--<artifactId>spark-catalyst_${project.scala.binary.version}</artifactId>-->
+      <!--<version>${dep.spark.version}</version>-->
+    <!--</dependency>-->
```
Comment (Mats-SX) [inline caps-core/pom.xml]: Remove these comments

File: caps-core/src/main/scala/org/opencypher/caps/api/record/CypherRecords.scala [hunk 3]
```diff
@@ -41,21 +48,21 @@ trait CypherRecords extends CypherPrintable {
...
+//    */
+//  def fields: Set[String]
+//
+//  /**
+//    * The explicitly named fields of this table, in the order they were defined.
+//    *
+//    * @return the sequence of fields of this table, in the order they were defined.
+//    * @see [[CypherRecordHeader.fieldsInOrder]]
+//    */
+//  def fieldsInOrder: Seq[String]
```
Comment (Mats-SX) [inline caps-core/src/main/scala/org/opencypher/caps/api/record/CypherRecords.scala]: Remove comments?

File: caps-spark/src/test/scala/org/opencypher/caps/api/schema/TestGraphSchemaTest.scala [hunk 4]
```diff
@@ -16,7 +16,7 @@
 package org.opencypher.caps.api.schema
 
 import org.opencypher.caps.api.types.{CTInteger, CTString}
-import org.opencypher.caps.test.CAPSTestSuite
+import org.opencypher.caps.test.{BaseTestSuite, CAPSTestSuite}
```
Comment (Mats-SX) [inline caps-spark/src/test/scala/org/opencypher/caps/api/schema/TestGraphSchemaTest.scala]: Looks unnecessary

File: caps-core/pom.xml (hunk 1 above)
Comment (DarthMax) [reply caps-core/pom.xml]: Nope, we found that in the Maven documentation. And it seems the version is inherited from the parent

File: caps-core/src/main/scala/org/opencypher/caps/api/graph/CypherSession.scala [hunk 5]
```diff
@@ -72,4 +72,8 @@ trait CypherSession {
   def graphAt(uri: URI): Graph
 }
 
+object CypherSession {
+  val sessionGraphSchema = "session"
```
Comment (DarthMax) [inline caps-core/src/main/scala/org/opencypher/caps/api/graph/CypherSession.scala]: sessionGraphScheme

File: caps-spark/src/main/scala/org/opencypher/caps/impl/spark/DfUtils.scala [hunk 6]
```diff
@@ -15,16 +15,17 @@
...
 import org.opencypher.caps.api.record.RecordHeader
+import org.opencypher.caps.api.spark.CAPSSession
 import org.opencypher.caps.api.types._
 import org.opencypher.caps.api.value._
 import org.opencypher.caps.api.value.instances._
 import org.opencypher.caps.impl.spark.exception.Raise
 import org.opencypher.caps.impl.spark.physical.RuntimeContext
 
-object RowUtils {
+object DfUtils {
```
Comment (DarthMax) [inline caps-spark/src/main/scala/org/opencypher/caps/impl/spark/DfUtils.scala]: rename to DataFrameUtils

**Pull request merged**